src/lib/hooks/useStaking.tsx

```diff
@@ -0,0 +1,210 @@
+import { useMemo, useCallback } from 'react';
+import { ethers } from 'ethers';
+import {
+	DomainRequestAndContents,
+	DomainRequestContents,
+	Maybe,
+	NftStatusCard,
+	StakingRequest,
+} from 'lib/types';
+import {
+	createDomainMetadata,
+	UploadedDomainMetadata,
+	uploadToIPFS,
+} from 'lib/utils';
+import useNotification from 'lib/hooks/useNotification';
+import { useStakingController } from 'lib/hooks/useStakingController';
+import { useStakingRedux } from 'store/staking/hooks';
+
+export type UseStakingReturn = {
```

Comment:
just a curious question mate...what is the difference using ```type``` over ```interface``` for the function props? I ask as I've been instructed in previous projects to use ```interface```

Reply:
@domw30 You are right. I sometimes forget it :(